### PR TITLE
sparse update with configurable threshold 

### PR DESCRIPTION
**Pull request by muqeeth**

### CODE REVIEW

File: git_theta/updates/sparse.py [hunk 1]
```diff
@@ -12,6 +13,13 @@
 class SparseUpdate(IncrementalUpdate):
     """An update where only some parameters are touched."""
 
+    def __init__(
+        self, serializer: params.Serializer, threshold: Optional[float] = 1e-12
```
Comment (blester125) [inline git_theta/updates/sparse.py:17]: `thereshold` shouldn't be `Optional` it is always used.

**Review by blester125 — CHANGES_REQUESTED** [review]
Can you add some quick tests? Things like:

* Given a consistent update value, increasing the threshold hold results in monotonically increasing sparseness
* Something that checks the sparseness of a known update? Like design one that will be 70% sparse after shareholding for example. 

You should be able to get sparseness ratios with `len(update_dict["values"]) / np.prod(param.shape)`

File: git_theta/updates/sparse.py (hunk 1 above)
Comment (muqeeth) [reply git_theta/updates/sparse.py]: Changed it to always used

**Comment by muqeeth** [comment]
Added above mentioned test cases

**Review by blester125 — APPROVED** [review]

**Pull request merged**